website/source/guides/configuration/authentication.html.md

```diff
@@ -0,0 +1,671 @@
+
+# Read the jenkins role
+$ vault read auth/approle/role/jenkins
+
+  Key               	Value
+  ---               	-----
+  bind_secret_id    	true
+  bound_cidr_list
+  period            	0
+  policies          	[jenkins]
+  secret_id_num_uses	0
+  secret_id_ttl     	0
+  token_max_ttl     	0
+  token_num_uses    	0
+  token_ttl         	0
+```
+
+**NOTE:** To attach multiple policies, pass the policy names as a comma
+separated string.
+
+```shell
+$ vault write auth/approle/role/jenkins policies="jenkins,anotherpolicy"
+````
+
+#### API call using cURL
+
+Before creating a role, create `jenkins` policy:
+
+```shell
+$ curl --header "X-Vault-Token: ..." --request PUT --data @payload.json \
+     https://vault.rocks/v1/sys/policy/jenkins
+
+$ cat payload.json
+{
+  "policy": "path \"auth/approle/login\" {  capabilities = [ \"create\", \"read\" ] } ... }"
+}
+```
+
+Now, you are ready to create a role.
+
+**Example:**
+
+The following example creates a role named `jenkins` with `jenkins` policy
+attached. (NOTE: This example creates a role operates in [**pull**
+mode](/docs/auth/approle.html).)
+
+```shell
+$ curl --header "X-Vault-Token: ..." --request POST \
+       --data '{"policies":"jenkins"}' \
+       https://vault.rocks/v1/auth/approle/role/jenkins
+```
+
+> There are a number of
+> [parameters](/api/auth/approle/index.html#create-new-approle) that you can set
+> on a role. If you want to limit the use of the generated secret ID, set
+> `secret_id_num_uses` or `secret_id_ttl` parameter values. Similarly, you can
+> specify `token_num_uses` and `token_ttl`. You may never want the app token to
+> expire.  In such a case, specify the `period` so that the token generated by
+> this AppRole is a periodic token.  To learn more about periodic token, refer to
+> the [Tokens and Leases](/guifes/lease.html#step4) guide.
+
+
+**NOTE:** To attach multiple policies, pass the policy names as a comma
+separated string.
+
+```shell
+$ curl --header "X-Vault-Token:..."
+       --request POST \
+       --data '{"policies":"jenkins,anotherpolicy"}' \
+       https://vault.rocks/v1/auth/approle/role/jenkins
+````
+
+To read the jenkins role you just created:
+
+```shell
+$ curl --header "X-Vault-Token: ..." --request GET \
+        https://vault.rocks/v1/auth/approle/role/jenkins | jq
+{
+  "request_id": "b18054ad-1ab5-8d83-eeed-193d97026ee7",
+  "lease_id": "",
+  "renewable": false,
+  "lease_duration": 0,
+  "data": {
+    "bind_secret_id": true,
+    "bound_cidr_list": "",
+    "period": 0,
+    "policies": [
+      "jenkins"
+    ],
+    "secret_id_num_uses": 0,
+    "secret_id_ttl": 0,
+    "token_max_ttl": 0,
+    "token_num_uses": 0,
+    "token_ttl": 0
+  },
+  "wrap_info": null,
+  "warnings": null,
+  "auth": null
+}
+```
+
+
+### <a name="step3"></a>Step 3: Get Role ID and Secret ID
+(**Persona:** admin)
+
+**Role ID** and **Secret ID** are like username and password which a machine or
+app uses to authenticate.
```

Comment:
are like _a_ username and password _that_ a machine or app uses to authenticate.